### PR TITLE
[CalendarService] Add sorting and searching API

### PR DESCRIPTION
**Pull request by VasiliyPetrov1**

### CODE REVIEW

File: src/main/java/ua/kpi/iasa/IASA_Organiser/service/EventDestributionService.java [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+import ua.kpi.iasa.IASA_Organiser.data.EventDistributionContainer;
+import ua.kpi.iasa.IASA_Organiser.data.GenericDataManager;
+import ua.kpi.iasa.IASA_Organiser.data.impl.DefaultListDataManager;
+import ua.kpi.iasa.IASA_Organiser.model.Event;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class EventDestributionService { // singleton
+    /* this class is Publisher in Observer pattern
```
Comment (MLGMAG) [inline src/main/java/ua/kpi/iasa/IASA_Organiser/service/EventDestributionService.java:12]: Add java docs to class with this comment. 
You can see example in PR#52

File: src/main/java/ua/kpi/iasa/IASA_Organiser/model/Calendar.java [hunk 2]
```diff
@@ -1,6 +1,8 @@
 package ua.kpi.iasa.IASA_Organiser.model;
 
-public class Calendar {
+import ua.kpi.iasa.IASA_Organiser.data.EventDistributionContainer;
+
+public class Calendar implements EventDistributionContainer {
     private Event[] events;
```
Comment (MLGMAG) [inline src/main/java/ua/kpi/iasa/IASA_Organiser/model/Calendar.java:6]: Recode Array to List

**Review by MLGMAG — CHANGES_REQUESTED** [review]

File: src/main/java/ua/kpi/iasa/IASA_Organiser/service/EventDestributionService.java [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    Subscribers are classes which inherit EventDestributionContainer interface
+    For instance Calendar is a Subscriber
+    */
+
+    private static EventDestributionService instance;
+    private GenericDataManager dataManager;
+    private List<EventDistributionContainer> subscribers;
+
```
Comment (makrushinoff) [inline src/main/java/ua/kpi/iasa/IASA_Organiser/service/EventDestributionService.java:20]: If you want to make singleton, you should create private constructor

File: src/main/java/ua/kpi/iasa/IASA_Organiser/model/Calendar.java (hunk 2 above)
Comment (VasiliyPetrov1) [reply src/main/java/ua/kpi/iasa/IASA_Organiser/model/Calendar.java:6]: Yes nice idea)

**Review by makrushinoff — CHANGES_REQUESTED** [review]

File: src/main/java/ua/kpi/iasa/IASA_Organiser/data/EventDistributionContainer.java [hunk 4]
```diff
@@ -0,0 +1,13 @@
+package ua.kpi.iasa.IASA_Organiser.data;
+
+import ua.kpi.iasa.IASA_Organiser.model.Event;
+
+public interface EventDistributionContainer {
```
Comment (MLGMAG) [inline src/main/java/ua/kpi/iasa/IASA_Organiser/data/EventDistributionContainer.java:5]: YAGNI

**Review by MLGMAG — CHANGES_REQUESTED** [review]

File: src/main/java/ua/kpi/iasa/IASA_Organiser/service/CalendarService.java [hunk 5]
```diff
@@ -1,59 +1,58 @@
...
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+import java.util.stream.Collectors;
 
-public class CalendarService {
+public class CalendarService{
     private static CalendarService instance;
-    private ArrayDataManager arrayDataManager = DefaultArrayDataManager.getInstance();
+    private GenericDataManager dataManager = DefaultFileDataManager.getInstance();
```
Comment (MLGMAG) [inline src/main/java/ua/kpi/iasa/IASA_Organiser/service/CalendarService.java:18]: Cool)

**Review by MLGMAG — APPROVED** [review]

**Review by MLGMAG — APPROVED** [review]

**Review by makrushinoff — APPROVED** [review]

**Pull request merged**